Unexpected error when using config file inheritance with project references

<!-- 🚨 STOP 🚨 STOP 🚨 STOP 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!--
Please try to reproduce the issue with the latest published version. It may have already been fixed.
For npm: `typescript@next`
This is also the 'Nightly' version in the playground: http://www.typescriptlang.org/play/?ts=Nightly
-->
**TypeScript Version:**  4.0.2

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** project references composite configuration file inheritance include files

**Code**

Reduced test case: https://github.com/OliverJAsh/ts-project-references-config-inheritance-bug. Below I have inlined the code from this repository.

Project structure:

```
├── shared
│   ├── index.ts
│   ├── tsconfig-base.json
│   ├── tsconfig.json (extends `shared/tsconfig-base.json`)
│   └── typings-base
│       └── globals.d.ts
├── webpack
│   ├── index.ts
│   └── tsconfig.json (extends `shared/tsconfig-base.json`)
└── tsconfig.json (references projects `shared/tsconfig.json` and `webpack/tsconfig.json`)
```

Contents:

`shared/index.ts`:
```ts
export const a: Unrestricted = 1;
```

`shared/tsconfig-base.json`:
```json
{
  "include": ["./typings-base/"]
}
```

`shared/tsconfig.json`:
```json
{
  "extends": "./tsconfig-base.json",
  "compilerOptions": {
    "composite": true,
    "outDir": "../target-tsc-build/",
    "rootDir": ".."
  },
  "files": ["./index.ts"]
}
```

`shared/typings-base/globals.d.ts`:
```ts
type Unrestricted = any;
```

`webpack/index.ts`:
```ts
export const b: Unrestricted = 1;
```

`webpack/tsconfig.json`:
```json
{
  "extends": "../shared/tsconfig-base.json",
  "compilerOptions": {
    "composite": true,
    "outDir": "../target-tsc-build/",
    "rootDir": ".."
  },
  "files": ["./index.ts"],
  "references": [
    {
      "path": "../shared/tsconfig.json"
    }
  ]
}
```

`tsconfig.json`:
```json
{
  "references": [
    {
      "path": "./shared/tsconfig.json"
    },
    {
      "path": "./webpack/tsconfig.json"
    }
  ],
  "files": []
}
```


**Expected behavior:**

```bash
# No errors ✅
tsc --build webpack/tsconfig.json --force --verbose

# No errors ✅
tsc --build tsconfig.json --force --verbose
```

**Actual behavior:**

```bash
# No errors ✅
tsc --build webpack/tsconfig.json --force --verbose

# Errors ❌
tsc --build tsconfig.json --force --verbose
```

The unexpected error:

```
[16:49:43] Projects in this build:
    * shared/tsconfig.json
    * webpack/tsconfig.json
    * tsconfig.json

[16:49:43] Project 'shared/tsconfig.json' is up to date because newest input 'shared/typings-base/globals.d.ts' is older than oldest output 'target-tsc-build/shared/index.js'

[16:49:43] Building project '/Users/oliverash/Development/ts-project-references-config-inheritance-bug/shared/tsconfig.json'...

[16:49:44] Project 'webpack/tsconfig.json' is out of date because oldest output 'target-tsc-build/webpack/index.js' is older than newest input 'shared/tsconfig.json'

[16:49:44] Building project '/Users/oliverash/Development/ts-project-references-config-inheritance-bug/webpack/tsconfig.json'...

webpack/index.ts:1:17 - error TS2304: Cannot find name 'Unrestricted'.

1 export const b: Unrestricted = 1;
                  ~~~~~~~~~~~~


Found 1 error.
```

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior -->

**Related Issues:** <!-- Did you find other bugs that looked similar? -->

This might not be related but I just ran into another similar case where again I'm seeing inconsistent errors when I run `tsc --build` with a project directly versus when run I run it with a project that references the aforementioned project.

https://github.com/OliverJAsh/ts-project-references-config-inheritance-bug/tree/problem-2

In this case, I'm expecting an error but not getting any.

```bash
# Expected error and got one ✅
tsc --build tsconfig.json --clean && \
    tsc --build tsconfig.json --verbose
```

As expected, this errors with:

```
error TS2318: Cannot find global type 'Array'.

error TS2318: Cannot find global type 'Boolean'.

error TS2318: Cannot find global type 'Function'.

error TS2318: Cannot find global type 'IArguments'.

error TS2318: Cannot find global type 'Number'.

error TS2318: Cannot find global type 'Object'.

error TS2318: Cannot find global type 'RegExp'.

error TS2318: Cannot find global type 'String'.


Found 8 errors.
```

However, if I run use the "solution" project instead, I don't get any errors:

```bash
# Expected error but got none ❌
tsc --build service-worker/tsconfig.json --clean && \
    tsc --build service-worker/tsconfig.json --verbose
```

--- a/src/compiler/commandLineParser.ts
+++ b/src/compiler/commandLineParser.ts
@@ -2617,14 +2617,17 @@ namespace ts {
         if (ownConfig.extendedConfigPath) {
             // copy the resolution stack so it is never reused between branches in potential diamond-problem scenarios.
             resolutionStack = resolutionStack.concat([resolvedPath]);
-            const extendedConfig = getExtendedConfig(sourceFile, ownConfig.extendedConfigPath, host, basePath, resolutionStack, errors, extendedConfigCache);
+            const extendedConfig = getExtendedConfig(sourceFile, ownConfig.extendedConfigPath, host, resolutionStack, errors, extendedConfigCache);
             if (extendedConfig && isSuccessfulParsedTsconfig(extendedConfig)) {
                 const baseRaw = extendedConfig.raw;
                 const raw = ownConfig.raw;
+                let relativeDifference: string | undefined ;
                 const setPropertyInRawIfNotUndefined = (propertyName: string) => {
-                    const value = raw[propertyName] || baseRaw[propertyName];
-                    if (value) {
-                        raw[propertyName] = value;
+                    if (!raw[propertyName] && baseRaw[propertyName]) {
+                        raw[propertyName] = map(baseRaw[propertyName], (path: string) => isRootedDiskPath(path) ? path : combinePaths(
+                            relativeDifference ||= convertToRelativePath(getDirectoryPath(ownConfig.extendedConfigPath!), basePath, createGetCanonicalFileName(host.useCaseSensitiveFileNames)),
+                            path
+                        ));
                     }
                 };
                 setPropertyInRawIfNotUndefined("include");
@@ -2786,7 +2789,6 @@ namespace ts {
         sourceFile: TsConfigSourceFile | undefined,
         extendedConfigPath: string,
         host: ParseConfigHost,
-        basePath: string,
         resolutionStack: string[],
         errors: Push<Diagnostic>,
         extendedConfigCache?: ESMap<string, ExtendedConfigCacheEntry>
@@ -2801,25 +2803,8 @@ namespace ts {
         else {
             extendedResult = readJsonConfigFile(extendedConfigPath, path => host.readFile(path));
             if (!extendedResult.parseDiagnostics.length) {
-                const extendedDirname = getDirectoryPath(extendedConfigPath);
-                extendedConfig = parseConfig(/*json*/ undefined, extendedResult, host, extendedDirname,
+                extendedConfig = parseConfig(/*json*/ undefined, extendedResult, host, getDirectoryPath(extendedConfigPath),
                     getBaseFileName(extendedConfigPath), resolutionStack, errors, extendedConfigCache);
-
-                if (isSuccessfulParsedTsconfig(extendedConfig)) {
-                    // Update the paths to reflect base path
-                    const relativeDifference = convertToRelativePath(extendedDirname, basePath, identity);
-                    const updatePath = (path: string) => isRootedDiskPath(path) ? path : combinePaths(relativeDifference, path);
-                    const mapPropertiesInRawIfNotUndefined = (propertyName: string) => {
-                        if (raw[propertyName]) {
-                            raw[propertyName] = map(raw[propertyName], updatePath);
-                        }
-                    };
-
-                    const { raw } = extendedConfig;
-                    mapPropertiesInRawIfNotUndefined("include");
-                    mapPropertiesInRawIfNotUndefined("exclude");
-                    mapPropertiesInRawIfNotUndefined("files");
-                }
             }
             if (extendedConfigCache) {
                 extendedConfigCache.set(path, { extendedResult, extendedConfig });

--- a/src/testRunner/tsconfig.json
+++ b/src/testRunner/tsconfig.json
@@ -112,6 +112,7 @@
         "unittests/services/transpile.ts",
         "unittests/tsbuild/amdModulesWithOut.ts",
         "unittests/tsbuild/configFileErrors.ts",
+        "unittests/tsbuild/configFileExtends.ts",
         "unittests/tsbuild/containerOnlyReferenced.ts",
         "unittests/tsbuild/declarationEmit.ts",
         "unittests/tsbuild/demo.ts",

--- a/src/testRunner/unittests/tsbuild/configFileExtends.ts
+++ b/src/testRunner/unittests/tsbuild/configFileExtends.ts
@@ -1,0 +1,52 @@
+namespace ts {
+    describe("unittests:: tsbuild:: configFileExtends:: when tsconfig extends another config", () => {
+        function getConfigExtendsWithIncludeFs() {
+            return loadProjectFromFiles({
+                "/src/tsconfig.json": JSON.stringify({
+                    references: [
+                        { path: "./shared/tsconfig.json" },
+                        { path: "./webpack/tsconfig.json" }
+                    ],
+                    files: []
+                }),
+                "/src/shared/tsconfig-base.json": JSON.stringify({
+                    include: ["./typings-base/"]
+                }),
+                "/src/shared/typings-base/globals.d.ts": `type Unrestricted = any;`,
+                "/src/shared/tsconfig.json": JSON.stringify({
+                    extends: "./tsconfig-base.json",
+                    compilerOptions: {
+                        composite: true,
+                        outDir: "../target-tsc-build/",
+                        rootDir: ".."
+                    },
+                    files: ["./index.ts"]
+                }),
+                "/src/shared/index.ts": `export const a: Unrestricted = 1;`,
+                "/src/webpack/tsconfig.json": JSON.stringify({
+                    extends: "../shared/tsconfig-base.json",
+                    compilerOptions: {
+                        composite: true,
+                        outDir: "../target-tsc-build/",
+                        rootDir: ".."
+                    },
+                    files: ["./index.ts"],
+                    references: [{ path: "../shared/tsconfig.json" }]
+                }),
+                "/src/webpack/index.ts": `export const b: Unrestricted = 1;`,
+            });
+        }
+        verifyTsc({
+            scenario: "configFileExtends",
+            subScenario: "when building solution with projects extends config with include",
+            fs: getConfigExtendsWithIncludeFs,
+            commandLineArgs: ["--b", "/src/tsconfig.json", "--v", "--listFiles"],
+        });
+        verifyTsc({
+            scenario: "configFileExtends",
+            subScenario: "when building project uses reference and both extend config with include",
+            fs: getConfigExtendsWithIncludeFs,
+            commandLineArgs: ["--b", "/src/webpack/tsconfig.json", "--v", "--listFiles"],
+        });
+    });
+}

--- a/tests/baselines/reference/tsbuild/configFileExtends/initial-build/when-building-project-uses-reference-and-both-extend-config-with-include.js
+++ b/tests/baselines/reference/tsbuild/configFileExtends/initial-build/when-building-project-uses-reference-and-both-extend-config-with-include.js
@@ -1,0 +1,160 @@
+Input::
+//// [/lib/lib.d.ts]
+/// <reference no-default-lib="true"/>
+interface Boolean {}
+interface Function {}
+interface CallableFunction {}
+interface NewableFunction {}
+interface IArguments {}
+interface Number { toExponential: any; }
+interface Object {}
+interface RegExp {}
+interface String { charAt: any; }
+interface Array<T> { length: number; [n: number]: T; }
+interface ReadonlyArray<T> {}
+declare const console: { log(msg: any): void; };
+
+//// [/src/shared/index.ts]
+export const a: Unrestricted = 1;
+
+//// [/src/shared/tsconfig-base.json]
+{"include":["./typings-base/"]}
+
+//// [/src/shared/tsconfig.json]
+{"extends":"./tsconfig-base.json","compilerOptions":{"composite":true,"outDir":"../target-tsc-build/","rootDir":".."},"files":["./index.ts"]}
+
+//// [/src/shared/typings-base/globals.d.ts]
+type Unrestricted = any;
+
+//// [/src/tsconfig.json]
+
+
+//// [/src/webpack/index.ts]
+export const b: Unrestricted = 1;
+
+//// [/src/webpack/tsconfig.json]
+{"extends":"../shared/tsconfig-base.json","compilerOptions":{"composite":true,"outDir":"../target-tsc-build/","rootDir":".."},"files":["./index.ts"],"references":[{"path":"../shared/tsconfig.json"}]}
+
+
+
+Output::
+/lib/tsc --b /src/webpack/tsconfig.json --v --listFiles
+[[90m12:00:00 AM[0m] Projects in this build: 
+    * src/shared/tsconfig.json
+    * src/webpack/tsconfig.json
+
+[[90m12:00:00 AM[0m] Project 'src/shared/tsconfig.json' is out of date because output file 'src/target-tsc-build/shared/index.js' does not exist
+
+[[90m12:00:00 AM[0m] Building project '/src/shared/tsconfig.json'...
+
+/lib/lib.d.ts
+/src/shared/index.ts
+/src/shared/typings-base/globals.d.ts
+[[90m12:00:00 AM[0m] Project 'src/webpack/tsconfig.json' is out of date because output file 'src/target-tsc-build/webpack/index.js' does not exist
+
+[[90m12:00:00 AM[0m] Building project '/src/webpack/tsconfig.json'...
+
+/lib/lib.d.ts
+/src/webpack/index.ts
+/src/shared/typings-base/globals.d.ts
+exitCode:: ExitStatus.Success
+
+
+//// [/src/target-tsc-build/shared/index.d.ts]
+export declare const a: Unrestricted;
+
+
+//// [/src/target-tsc-build/shared/index.js]
+"use strict";
+exports.__esModule = true;
+exports.a = void 0;
+exports.a = 1;
+
+
+//// [/src/target-tsc-build/shared/tsconfig.tsbuildinfo]
+{
+  "program": {
+    "fileInfos": {
+      "../../../lib/lib.d.ts": {
+        "version": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "signature": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "affectsGlobalScope": true
+      },
+      "../../shared/index.ts": {
+        "version": "-22125360210-export const a: Unrestricted = 1;",
+        "signature": "-478734393-export declare const a: Unrestricted;\r\n",
+        "affectsGlobalScope": false
+      },
+      "../../shared/typings-base/globals.d.ts": {
+        "version": "4725476611-type Unrestricted = any;",
+        "signature": "4725476611-type Unrestricted = any;",
+        "affectsGlobalScope": true
+      }
+    },
+    "options": {
+      "composite": true,
+      "outDir": "..",
+      "rootDir": "../..",
+      "listFiles": true,
+      "configFilePath": "../../shared/tsconfig.json"
+    },
+    "referencedMap": {},
+    "exportedModulesMap": {},
+    "semanticDiagnosticsPerFile": [
+      "../../../lib/lib.d.ts",
+      "../../shared/index.ts",
+      "../../shared/typings-base/globals.d.ts"
+    ]
+  },
+  "version": "FakeTSVersion"
+}
+
+//// [/src/target-tsc-build/webpack/index.d.ts]
+export declare const b: Unrestricted;
+
+
+//// [/src/target-tsc-build/webpack/index.js]
+"use strict";
+exports.__esModule = true;
+exports.b = void 0;
+exports.b = 1;
+
+
+//// [/src/target-tsc-build/webpack/tsconfig.tsbuildinfo]
+{
+  "program": {
+    "fileInfos": {
+      "../../../lib/lib.d.ts": {
+        "version": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "signature": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "affectsGlobalScope": true
+      },
+      "../../webpack/index.ts": {
+        "version": "-14405273073-export const b: Unrestricted = 1;",
+        "signature": "-5074241048-export declare const b: Unrestricted;\r\n",
+        "affectsGlobalScope": false
+      },
+      "../../shared/typings-base/globals.d.ts": {
+        "version": "4725476611-type Unrestricted = any;",
+        "signature": "4725476611-type Unrestricted = any;",
+        "affectsGlobalScope": true
+      }
+    },
+    "options": {
+      "composite": true,
+      "outDir": "..",
+      "rootDir": "../..",
+      "listFiles": true,
+      "configFilePath": "../../webpack/tsconfig.json"
+    },
+    "referencedMap": {},
+    "exportedModulesMap": {},
+    "semanticDiagnosticsPerFile": [
+      "../../../lib/lib.d.ts",
+      "../../shared/typings-base/globals.d.ts",
+      "../../webpack/index.ts"
+    ]
+  },
+  "version": "FakeTSVersion"
+}
+

--- a/tests/baselines/reference/tsbuild/configFileExtends/initial-build/when-building-solution-with-projects-extends-config-with-include.js
+++ b/tests/baselines/reference/tsbuild/configFileExtends/initial-build/when-building-solution-with-projects-extends-config-with-include.js
@@ -55,17 +55,10 @@ Output::
 
 [[90m12:00:00 AM[0m] Building project '/src/webpack/tsconfig.json'...
 
-[96msrc/webpack/index.ts[0m:[93m1[0m:[93m17[0m - [91merror[0m[90m TS2304: [0mCannot find name 'Unrestricted'.
-
-[7m1[0m export const b: Unrestricted = 1;
-[7m [0m [91m                ~~~~~~~~~~~~[0m
-
 /lib/lib.d.ts
 /src/webpack/index.ts
-
-Found 1 error.
-
-exitCode:: ExitStatus.DiagnosticsPresent_OutputsGenerated
+/src/shared/typings-base/globals.d.ts
+exitCode:: ExitStatus.Success
 
 
 //// [/src/target-tsc-build/shared/index.d.ts]
@@ -117,6 +110,17 @@ exports.a = 1;
   "version": "FakeTSVersion"
 }
 
+//// [/src/target-tsc-build/webpack/index.d.ts]
+export declare const b: Unrestricted;
+
+
+//// [/src/target-tsc-build/webpack/index.js]
+"use strict";
+exports.__esModule = true;
+exports.b = void 0;
+exports.b = 1;
+
+
 //// [/src/target-tsc-build/webpack/tsconfig.tsbuildinfo]
 {
   "program": {
@@ -130,6 +134,11 @@ exports.a = 1;
         "version": "-14405273073-export const b: Unrestricted = 1;",
         "signature": "-5074241048-export declare const b: Unrestricted;\r\n",
         "affectsGlobalScope": false
+      },
+      "../../shared/typings-base/globals.d.ts": {
+        "version": "4725476611-type Unrestricted = any;",
+        "signature": "4725476611-type Unrestricted = any;",
+        "affectsGlobalScope": true
       }
     },
     "options": {
@@ -143,25 +152,8 @@ exports.a = 1;
     "exportedModulesMap": {},
     "semanticDiagnosticsPerFile": [
       "../../../lib/lib.d.ts",
-      [
-        "../../webpack/index.ts",
-        [
-          {
-            "file": "../../webpack/index.ts",
-            "start": 16,
-            "length": 12,
-            "messageText": "Cannot find name 'Unrestricted'.",
-            "category": 1,
-            "code": 2304
-          }
-        ]
-      ]
-    ],
-    "affectedFilesPendingEmit": [
-      [
-        "../../webpack/index.ts",
-        1
-      ]
+      "../../shared/typings-base/globals.d.ts",
+      "../../webpack/index.ts"
     ]
   },
   "version": "FakeTSVersion"

--- a/tests/baselines/reference/tsbuild/configFileExtends/initial-build/when-building-solution-with-projects-extends-config-with-include.js
+++ b/tests/baselines/reference/tsbuild/configFileExtends/initial-build/when-building-solution-with-projects-extends-config-with-include.js
@@ -1,0 +1,169 @@
+Input::
+//// [/lib/lib.d.ts]
+/// <reference no-default-lib="true"/>
+interface Boolean {}
+interface Function {}
+interface CallableFunction {}
+interface NewableFunction {}
+interface IArguments {}
+interface Number { toExponential: any; }
+interface Object {}
+interface RegExp {}
+interface String { charAt: any; }
+interface Array<T> { length: number; [n: number]: T; }
+interface ReadonlyArray<T> {}
+declare const console: { log(msg: any): void; };
+
+//// [/src/shared/index.ts]
+export const a: Unrestricted = 1;
+
+//// [/src/shared/tsconfig-base.json]
+{"include":["./typings-base/"]}
+
+//// [/src/shared/tsconfig.json]
+{"extends":"./tsconfig-base.json","compilerOptions":{"composite":true,"outDir":"../target-tsc-build/","rootDir":".."},"files":["./index.ts"]}
+
+//// [/src/shared/typings-base/globals.d.ts]
+type Unrestricted = any;
+
+//// [/src/tsconfig.json]
+{"references":[{"path":"./shared/tsconfig.json"},{"path":"./webpack/tsconfig.json"}],"files":[]}
+
+//// [/src/webpack/index.ts]
+export const b: Unrestricted = 1;
+
+//// [/src/webpack/tsconfig.json]
+{"extends":"../shared/tsconfig-base.json","compilerOptions":{"composite":true,"outDir":"../target-tsc-build/","rootDir":".."},"files":["./index.ts"],"references":[{"path":"../shared/tsconfig.json"}]}
+
+
+
+Output::
+/lib/tsc --b /src/tsconfig.json --v --listFiles
+[[90m12:00:00 AM[0m] Projects in this build: 
+    * src/shared/tsconfig.json
+    * src/webpack/tsconfig.json
+    * src/tsconfig.json
+
+[[90m12:00:00 AM[0m] Project 'src/shared/tsconfig.json' is out of date because output file 'src/target-tsc-build/shared/index.js' does not exist
+
+[[90m12:00:00 AM[0m] Building project '/src/shared/tsconfig.json'...
+
+/lib/lib.d.ts
+/src/shared/index.ts
+/src/shared/typings-base/globals.d.ts
+[[90m12:00:00 AM[0m] Project 'src/webpack/tsconfig.json' is out of date because output file 'src/target-tsc-build/webpack/index.js' does not exist
+
+[[90m12:00:00 AM[0m] Building project '/src/webpack/tsconfig.json'...
+
+[96msrc/webpack/index.ts[0m:[93m1[0m:[93m17[0m - [91merror[0m[90m TS2304: [0mCannot find name 'Unrestricted'.
+
+[7m1[0m export const b: Unrestricted = 1;
+[7m [0m [91m                ~~~~~~~~~~~~[0m
+
+/lib/lib.d.ts
+/src/webpack/index.ts
+
+Found 1 error.
+
+exitCode:: ExitStatus.DiagnosticsPresent_OutputsGenerated
+
+
+//// [/src/target-tsc-build/shared/index.d.ts]
+export declare const a: Unrestricted;
+
+
+//// [/src/target-tsc-build/shared/index.js]
+"use strict";
+exports.__esModule = true;
+exports.a = void 0;
+exports.a = 1;
+
+
+//// [/src/target-tsc-build/shared/tsconfig.tsbuildinfo]
+{
+  "program": {
+    "fileInfos": {
+      "../../../lib/lib.d.ts": {
+        "version": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "signature": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "affectsGlobalScope": true
+      },
+      "../../shared/index.ts": {
+        "version": "-22125360210-export const a: Unrestricted = 1;",
+        "signature": "-478734393-export declare const a: Unrestricted;\r\n",
+        "affectsGlobalScope": false
+      },
+      "../../shared/typings-base/globals.d.ts": {
+        "version": "4725476611-type Unrestricted = any;",
+        "signature": "4725476611-type Unrestricted = any;",
+        "affectsGlobalScope": true
+      }
+    },
+    "options": {
+      "composite": true,
+      "outDir": "..",
+      "rootDir": "../..",
+      "listFiles": true,
+      "configFilePath": "../../shared/tsconfig.json"
+    },
+    "referencedMap": {},
+    "exportedModulesMap": {},
+    "semanticDiagnosticsPerFile": [
+      "../../../lib/lib.d.ts",
+      "../../shared/index.ts",
+      "../../shared/typings-base/globals.d.ts"
+    ]
+  },
+  "version": "FakeTSVersion"
+}
+
+//// [/src/target-tsc-build/webpack/tsconfig.tsbuildinfo]
+{
+  "program": {
+    "fileInfos": {
+      "../../../lib/lib.d.ts": {
+        "version": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "signature": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
+        "affectsGlobalScope": true
+      },
+      "../../webpack/index.ts": {
+        "version": "-14405273073-export const b: Unrestricted = 1;",
+        "signature": "-5074241048-export declare const b: Unrestricted;\r\n",
+        "affectsGlobalScope": false
+      }
+    },
+    "options": {
+      "composite": true,
+      "outDir": "..",
+      "rootDir": "../..",
+      "listFiles": true,
+      "configFilePath": "../../webpack/tsconfig.json"
+    },
+    "referencedMap": {},
+    "exportedModulesMap": {},
+    "semanticDiagnosticsPerFile": [
+      "../../../lib/lib.d.ts",
+      [
+        "../../webpack/index.ts",
+        [
+          {
+            "file": "../../webpack/index.ts",
+            "start": 16,
+            "length": 12,
+            "messageText": "Cannot find name 'Unrestricted'.",
+            "category": 1,
+            "code": 2304
+          }
+        ]
+      ]
+    ],
+    "affectedFilesPendingEmit": [
+      [
+        "../../webpack/index.ts",
+        1
+      ]
+    ]
+  },
+  "version": "FakeTSVersion"
+}
+

